### DISCUSSION TITLE
Non-traversable Roadstop Tiles

### DISCUSSION THREAD

**Discussion by KeepinItRail** [OpenTTD/OpenTTD#13186] [Features]
Could we make it so that Roadstop Stations have a way, similarly to Rail stations, to make a tile non-traversable, e.g. by having a flag (RST_GENERAL_FLAG_NON_TRAVERSABLE?).

The NML spec currently includes functionality, such as not allowing catenary on a Roadstop, stopping road pieces from auto-connecting, however it is still possible for a player to get a road vehicle to pass through such a station.

![image](https://github.com/user-attachments/assets/6a477a24-a30d-4f53-bfed-dfb4cf0276dd)

Would it be possible to have a method to prevent this, to have 'fully functional' Roadstop station buildings?

**Comment by yozora3** [comment]
I'd like that, would be really useful for creating large "hubs" without constant changing from/to trackless train stations

**Comment by Ufiby** [comment]
NewGRF: U&RaTT:RoadStops 0.1
Some stops have flags set up "RST_GENERAL_FLAG_BUILD_MENU_ROAD_ONLY"
It's just for show in the window for the road only, it doesn't show on the tram, but you can get around it. First, build stops, then build a tram and this one
![GIF 23 12 2024 16-29-45](https://github.com/user-attachments/assets/e26379b2-ce3f-4f27-ad81-0ae59ee0a7d4)
I wanted to ban the construction of a tram if the flag is set exclusively for roads. Would it be possible to find a way to prevent this?
